tecton_client/_internal/data_types.py

```diff
@@ -45,16 +47,40 @@ def get_features_dict(self) -> Dict[str, Any]:
             for metadata, raw_feature_value in zip(features_metadata, self.result.features)
         }
 
-    def _get_feature_value(self, data_type, raw_feature_value):
+    def _get_feature_value(self, data_type_info, raw_feature_value):
```

Comment:
This seems closer in semantics to a pure function right? We don't use `self.result` nor `self.metadata`. Let's move it out of the class and then we leave ourselves open for better unit tests

Reply:
like ideally we had unit tests that just loops over a set of possible data types and json and test those directly

Reply:
Will make the function static; that's equivalent, but still seems like it belongs in the GetFeaturesResponse namespace. Added tests.